ramips: mt7620: various minor wireless issue fixes for some devices

DragonBluep opened the pull request:
This patch series fixes some minor wireless issue for mt7620:

ramips: mt7620: add missing PA/LNA pinctrl for Linksys E1700
Fixes: https://github.com/openwrt/openwrt/issues/7959

mac80211: rt2x00: fix eeprom load from PCI eFuse 
Fixes: https://github.com/openwrt/openwrt/issues/17854

base-files: fix offset conversion on caldata_patch_data()
Revert "Revert "ramips: mt7620: fix patching mac address in caldata"" 
Fixes: https://github.com/openwrt/openwrt/issues/17818

DragonBluep committed:
mac80211: rt2x00: fix eeprom load from PCI eFuse

When eeprom name is not defined, the file load function should
return an error code so that it can fallthrough to read eeprom
form NIC inside eFuse.

Fixes: https://github.com/openwrt/openwrt/issues/17854
Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Link: https://github.com/openwrt/openwrt/pull/17892
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

hauke commented:
Please add a 
```
Fixes: https://github.com/openwrt/openwrt/issues/17854
```

I reverted a change in https://github.com/openwrt/openwrt/commit/150aa9aeee509dba5255885618d8df372670e8f3 to fix https://github.com/openwrt/openwrt/issues/17818, but I am not sure if it really fixed the problem. Do you have any of these devices or have an idea what happened there?

DragonBluep committed:
ramips: mt7620: add missing PA/LNA pinctrl for Linksys E1700

This device has high-power external PA and LNA[1]. So we must
configure PA/LNA pins to make the wireless work properly.

[1] https://fccid.io/Q87-E1700/Internal-Photos/Internal-Photos-pdf-2135639.pdf
Fixes: https://github.com/openwrt/openwrt/issues/7959
Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Link: https://github.com/openwrt/openwrt/pull/17892
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
base-files: fix offset conversion on caldata_patch_data()

`$(( ))` will convert uninitialized variable to "0". If we want to
use "-n" to check the string length, it's necessary to make sure the
converted variable is not empty.

Fixes: 652a6677d5fa ("base-files: Add new functions for ath11k caldata")
Fixes: https://github.com/openwrt/openwrt/issues/17818
Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Link: https://github.com/openwrt/openwrt/pull/17892
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
Revert "Revert "ramips: mt7620: fix patching mac address in caldata""

This reverts commit f628467dfd6b32ead172d33a2593f04901395343.

The initial fix was correct. However, a recently introduced bug in
base-files can cause some unexpected byte overwriting in eeprom.
Since it has been fixed, let's accept this patch again.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Link: https://github.com/openwrt/openwrt/pull/17892
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep commented:
> Please add a
> 
> ```
> Fixes: https://github.com/openwrt/openwrt/issues/17854
> ```

Done. 

> 
> I reverted a change in [150aa9a](https://github.com/openwrt/openwrt/commit/150aa9aeee509dba5255885618d8df372670e8f3) to fix #17818, but I am not sure if it really fixed the problem. 

Yes, it can fix this problem. But it is not the root cause of the problem. I found there is a bug in base-files which will touch some unexpected eeprom bytes used to verify the NIC model. I've pushed a new commit to solve it.
```
root@OpenWrt:~# echo $(())
0
root@OpenWrt:~# echo $(( ))
0
root@OpenWrt:~# echo $(($nothing))
0
```


I also added a commit trying to solve the wireless issue for Linksys E1700. I'm not sure if it can really fix the report https://github.com/openwrt/openwrt/issues/17818#issuecomment-2640034373, but it's clear that it will have some improvements. @589userr589

589userr589 commented:
I have a Linksys E1700, how can I help?

hauke commented:
@589userr589  Please install these changes on a Linksys E1700 and check if the wifi is stringer than without these changes.

589userr589 commented:
Ok how do I apply the changes? use snapchat image?
I'm going to need a little more explanation or help here.

stephrette313 reviewed (commented):
This is the owner of my hacked website & you are being defrauded please stop & report what you know to your authorities for fraud & identity theft plus Hippa violations & civil rights violation

DragonBluep commented:
> I have a Linksys E1700, how can I help?

@589userr589 Please test this one.
https://downloads.openwrt.org/releases/24.10-SNAPSHOT/targets/ramips/mt7620/openwrt-24.10-snapshot-r28459-b7b6ae7424-ramips-mt7620-linksys_e1700-squashfs-sysupgrade.bin

589userr589 commented:
76 / 5,000
Team, what's up with the message from "stephrette313"? Does anyone have any information on how and what?

589userr589 commented:
Team, good news. YOU DID IT. Well done and thank you.

I tested on version: OpenWrt GCC 13.3.0 r28832-74f0c95154

I tested the following options:
- Router or AP mode works,
- Wifi client works,
- Wifi-WPA2 encryption at 20Mhz channel width has communication throughput 40 - 60Mbsp (flowd - none)
- Wifi-WPA2 encryption at 40Mhz channel width has communication throughput 60 - 80Mbsp (flowd - none)
- Wifi-WPA2 encryption at 20Mhz channel width has communication throughput 60 - 80Mbsp (flowd - hw/sw)
- Wifi-WPA2 encryption at 40Mhz channel width has communication throughput 80 - 110Mbsp (flowd - hw/sw)
- Wifi-WPA3 encryption at 20Mhz channel width has communication throughput ~ 10Mbsp (flowd - none)
- Wifi-WPA3 encryption at a channel width of 40Mhz has a communication throughput of ~ 10Mbsp (flowd - none)
- Lan to Wan has a communication throughput of 140 - 160Mbsp (flowd - none)
- Lan to Wan has a communication throughput of at least 500Mbsp (flowd - hw/sw)
